functional-samples/cookbook.geolocation-offscreen/service_worker.js

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const OFFSCREEN_DOCUMENT_PATH = '/offscreen.html';
+
+chrome.runtime.onMessage.addListener(handleMessages);
```

Comment:
It'd be nice to add a high-level function `getGeoLocation` here that returns the current position. I think it'd be good to also provide a snippet for the promise juggling that this requires.

Reply:
Are you suggesting removing the current general `handleMessages` approach, and replace it with a `getGeolocation`, or supplementing the existing code with a convenience `getGeolocation`?

For the later, its a bit awkward, but the simplest addition to the existing code would look like

```js
function getGeolocation() {
  return new Promise(async (resolve, reject) => {
    
    chrome.runtime.onMessage.addListener(msg => {
      if (msg.type === 'geolocation-success') {
        resolve(msg.data);
      } else if (msg.type === 'geolocation-err') {
        reject(`unable to get geolocation data - ${msg.data}`)
      }
    })
    
    await sendMessageToOffscreenDocument('get-geolocation')
  })
}
```


Happy to do the former, just wanted to check first

Reply:
Afaik the later is not recommended as it nests callback registration in a method call. The former would be better imo.

Reply:
yeah, I don't think there is a way around nesting, as you need to have the reference to the `resolve` of the initiator. I'll refactor